iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/naming/jndi/CNCtx.java

```diff
@@ -406,27 +408,52 @@ private String getStringifiedIor(String url) throws NamingException {
                     String str;
                     while ((str = bufin.readLine()) != null) {
                         if (str.startsWith("IOR:")) {
-                            return str;
+                            retValue = str;
+                            break;
                         }
                     }
                 }
             } catch (IOException e) {
                 NamingException ne = IIOPLogger.ROOT_LOGGER.invalidURLOrIOR(url);
                 ne.setRootCause(e);
-                throw ne;
+                originalException = ne;
+            } catch (Throwable ex) {
+                if (ex instanceof InterruptedException) {
+                    Thread.currentThread().interrupt();
+                }
+                originalException = ex;
             } finally {
-                try {
-                    if (in != null) {
+                Throwable suppressed = originalException; // OK to be null
+                if (in != null) {
+                    try {
                         in.close();
+                    } catch (IOException e) {
+                        NamingException ne = IIOPLogger.ROOT_LOGGER.invalidURLOrIOR(url);
+                        ne.setRootCause(e);
+                        if (suppressed != null) {
+                            suppressed.addSuppressed(ne);
+                        } else {
+                            suppressed = ne;
+                        }
                     }
-                } catch (IOException e) {
-                    NamingException ne = IIOPLogger.ROOT_LOGGER.invalidURLOrIOR(url);
-                    ne.setRootCause(e);
-                    throw ne;
                 }
+                if (suppressed != null) {
+                    if (suppressed instanceof RuntimeException) {
```

Comment:
Maybe we could simplify it a bit and just do or and rethrow. Why do we need the cast?

Reply:
@tadamski in general this "pattern" is the way I solved all similar cases. In new Line 420 the Throwable may contain the RuntimeException or Error. You can't rethrow a undeclared Throwable, therefor the cast.

Reply:
@tadamski Christian Beikov had the same concern at the same point. He refactored it this way: [https://github.com/hibernate/hibernate-orm/blob/wip/6.0/hibernate-core/src/main/java/org/hibernate/internal/util/ExceptionHelper.java](https://github.com/hibernate/hibernate-orm/blob/wip/6.0/hibernate-core/src/main/java/org/hibernate/internal/util/ExceptionHelper.java) It works, but it completely undermines the typesafety of throw / Throwables. I don't want to provide it without broader review/acceptance.